skada/_utils.py

```diff
@@ -40,30 +51,87 @@ def _estimate_covariance(X, shrinkage):
 def check_X_y_domain(
     X,
     y,
-    sample_domain,
+    sample_domain=None,
     allow_source: bool = True,
     allow_multi_source: bool = True,
     allow_target: bool = True,
     allow_multi_target: bool = True,
     return_indices: bool = False,
     # xxx(okachaiev): most likely this needs to be removed as it doesn't fit new API
```

Comment:
if we shoudl remove something aybe do it while we are working on ythis function?

Reply:
I bet there's a separate task for this)